### PR TITLE
[Config] Introduce BuilderAwareInterface

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no     
| Deprecations? | no
| Tests pass?   | yes    
| Fixed tickets | #...   
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

Split `ParentNodeDefinitionInterface` into `BuilderAwareInterface`.

Use case is custom node definition  (extended from VariableNodeDef) with a corresponding prototyped array node.

To set the actual prototype i need the builder at definition level, provided by `ParentNodeDefinitionInterface`. However i don't implement `children()` + `append()`, i solely need the builder scope.

To go after #26297

### CODE REVIEW

File: src/Symfony/Component/Config/Definition/Builder/BuilderAwareInterface.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Definition\Builder;
+
+/**
+ * An interface that can be implemented by nodes which built other nodes.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Config/Definition/Builder/BuilderAwareInterface.php]: s/built/build

**Review by nicolas-grekas — APPROVED** [review]
with minor comment

**Review by 20uf — APPROVED** [review]

File: src/Symfony/Component/Config/Definition/Builder/BuilderAwareInterface.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * An interface that can be implemented by nodes which build other nodes.
+ *
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+interface BuilderAwareInterface
+{
+    /**
+     * Sets a custom children builder.
```
Comment (Simperfit) [inline src/Symfony/Component/Config/Definition/Builder/BuilderAwareInterface.php]: is this really needed :p ?

Comment (Simperfit) [reply src/Symfony/Component/Config/Definition/Builder/BuilderAwareInterface.php]: It can be just comment, and not phpdoc IMHO.

**Review by Simperfit — APPROVED** [review]
with a really minor comment ^^.

File: src/Symfony/Component/Config/Definition/Builder/BuilderAwareInterface.php (hunk 2 above)
Comment (ro0NL) [reply src/Symfony/Component/Config/Definition/Builder/BuilderAwareInterface.php]: I only moved things around, https://github.com/symfony/symfony/pull/26335/files#diff-b579be54110e0018ae82a55725bc373fL49

works for me as is :)

Comment (chalasr) [reply src/Symfony/Component/Config/Definition/Builder/BuilderAwareInterface.php]: We use phpdoc for documenting methods, not wild comments. So it's either a docblock or nothing at all, personally I would remove it as it does not bring much IMHO.

**Review by chalasr — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.

**Pull request closed without merging**